crates/networking/rpc/based/env.rs

```diff
@@ -0,0 +1,102 @@
+use ethrex_common::{Address, H256, U256};
+use serde::{Deserialize, Serialize};
+use ssz_types::{typenum, VariableList};
+use tree_hash::TreeHash;
+
+use crate::{utils::RpcErr, RpcApiContext};
+
+pub type MaxExtraDataSize = typenum::U256;
+pub type ExtraData = VariableList<u8, MaxExtraDataSize>;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EnvV0 {
+    pub number: u64,
+    pub parent_hash: H256,
+    pub beneficiary: Address,
+    pub timestamp: u64,
+    pub gas_limit: u64,
+    #[serde(rename = "baseFee")]
+    pub basefee: u64,
+    pub difficulty: U256,
+    pub prevrandao: H256,
+    #[serde(with = "ssz_types::serde_utils::hex_var_list")]
+    pub extra_data: ExtraData,
+    pub parent_beacon_block_root: H256,
+}
+
+impl EnvV0 {
+    pub fn handle(&self, _context: RpcApiContext) -> Result<serde_json::Value, RpcErr> {
+        tracing::info!("handling env");
+        Ok(serde_json::Value::Null)
+    }
+}
+
+impl TreeHash for EnvV0 {
+    fn tree_hash_type() -> tree_hash::TreeHashType {
+        tree_hash::TreeHashType::Container
+    }
+
+    fn tree_hash_packed_encoding(&self) -> tree_hash::PackedEncoding {
+        unreachable!("Struct should never be packed.")
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        unreachable!("Struct should never be packed.")
+    }
+
+    fn tree_hash_root(&self) -> tree_hash::Hash256 {
+        let mut hasher = tree_hash::MerkleHasher::with_leaves(10);
```

Comment:
In which cases can the hasher fail?

Reply:
Done the refactor you suggested in Slack [2ca7baf](https://github.com/lambdaclass/ethrex/pull/2022/commits/2ca7bafe3351fab2f13047adbe978048b10bcf3c).